apps/dashr-opiod-endemic/app.R

```diff
@@ -0,0 +1,383 @@
+                                x = 0.95,
+                                y = 0.85 - ((i-1) / 20),
+                                ax = -60,
+                                ay = 0,
+                                arrowwidth = 5,
+                                arrowhead = 0,
+                                bgcolor = "#1f2630",
+                                font = list(color = "#2cfec1")
+                              )
+                            })
+    
+    annotations <- append(annotations, annotations_1)
+    
+    if("layout" %in% figure){
+      lat = figure[["layout"]][["mapbox"]][["center"]]["lat"]  
+      lon = figure[["layout"]][["mapbox"]][["center"]]["lon"] 
+      zoom = figure[["layout"]][["mapbox"]]["zoom"]  
+    } else{
+      lat = 38.72490
+      lon = -95.61446
+      zoom = 3.5
+    }
+    
+    layout <- list(
+      mapbox = list(
+        layers = list(),
+        accesstoken = mapbox_access_token,
+        style = mapbox_style,
+        center = list(lat = lat, lon = lon),
+        zoom = zoom
+      ),
+      hovermode = "closet",
+      margin = list(r = 0, l = 0, t = 0, b = 0),
+      annotations = annotations,
+      dragmode = "lasso"
+    )
+    
+    base_url = "https://raw.githubusercontent.com/jackparmer/mapbox-counties/master/"
+    
+    ### CHeck this function
+    cm_name <- sapply(cm, function(n) n[1])
+    geo_layer <- lapply(1:length(BINS),
+                        function(i){
+                          index <- which(cm_name %in% BINS[[i]])
+                          color <- cm[[index]][2]
+                          list(
+                            sourcetype = "geojson",
+                            source = paste0(base_url, as.character(year), "/", BINS[[i]], ".geojson"),
+                            type = "fill",
+                            color = color,
+                            opacity = DEFAULT_OPACITY,
+                            fill = list(outlinecolor = "#afafaf")
+                          )
+                        })
+    
+    layout$mapbox$layers = geo_layer 
+    
+    fig <- list("data" = data, "layout" = layout)
+    
+    return(fig)
+    
+  }
+)
+
+app$callback(
+  output = list("id" = "heatmap-title", property = "children"),
+  params = list(input("id" = "years-slider", property = "value")),
+  
+  function(year){
+    # Functio prints year selected from range slider
+    return(glue("Heatmap of age adjusted mortality rates from poisonings in year {year}"))
+  }
+  
+)
+
+app$callback(
+  output = list(id = "selected-data", property = "figure"),
+  params = list(input(id = "county-choropleth", property = "selectedData"), 
+                input(id = "chart-dropdown", property = "value"),
+                input(id = "years-slider", property = "value")),
+  
+  function(selectedData, chart_dropdown, year){
+    # function returns blank figure if no data is selected
+    if(is.null(selectedData[[1]])){
+      print("SelectedData is None")
```

Comment:
This is probably for your personal debugging, probably can be removed.

Maybe if there is no selected data, you could add a default county or counties so that there isn't an empty histogram? Or some sort of message on the app itself prompting the user to select a county?